File: react-type-vite/src/services/api/response/expertResponse.ts
export interface ExpertResponse {
    id: string;
    username: string;
    email: string;
    firstName: string;
    lastName: string;
    expertId: string;
    educationalUnitId: string;
    description?: string;
    accountStatus: string;
    avatarUrl?: string;
    phoneNumber?: string;
    bio?: string;
    // department?: DepartmentResponse; // Experts might not have department
}

export interface ExpertRequest {
    username: string;
    password?: string;
    email: string;
    firstName: string;
    lastName: string;
    expertId: string;
    description?: string;
    educationalUnitId?: string;
    accountStatus?: string;
}

File: spring-microservices/identity-service/src/main/java/com/devteria/identity/dto/response/ExpertResponse.java
package com.devteria.identity.dto.response;

import lombok.*;
import lombok.experimental.FieldDefaults;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class ExpertResponse {
    private String id;

    private String username;

    private String email;

    private String firstName;

    private String lastName;

    private String dob;

    private String expertId;

    private String educationalUnitId;

    private String description;

    private String accountStatus;

    private String avatarUrl;

    private String phoneNumber;

    private String bio;
}
